src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let copyState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+      resultArray.push(Object.assign({}, copyState));
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+      resultArray.push(Object.assign({}, copyState));
+    } else if (action.type === 'clear') {
+      copyState = {};
+      resultArray.push({});
+    }
```

Comment:
[CODE STYLE]: Replace the `if...else if` structure with a `switch` statement to handle different action types. This will make the code more readable and better organized when dealing with a limited set of conditions.